<img style="float: left; margin: 30px 15px 15px 15px;" src="https://pngimage.net/wp-content/uploads/2018/06/logo-iteso-png-5.png" width="300" height="500" /> 
    
    
### <font color='navy'> Simulación de procesos financieros. 

**Nombres:** Sean Nicolás González Vázquez y Doris Gomez Contreras.

**Fecha:** 5 de noviembre del 2020.

**Expediente** : 720083, 713897.
**Profesor:** Oscar David Jaramillo Zuluaga.

# Tarea 9.

### El link donde se evidencia el trabajo es el siguiente: [GitHub Repositorio Tarea 9](https://github.com/nicoglez/Tarea9_SNGonzalez_DGomez)

In [2]:
# importacion de librerias
import numpy as np
import pandas as pd
import scipy.stats as st

## Enunciado de tarea 

# Ejercicio 1: prueba de chi cuadrada de independencia

Para ver si la venta de chips de silicio son independientes del punto del ciclo de negocios en que se encuentre la economía del país se han recogido las ventas semanales de una empresa y datos acerca de la economía del país, y se reportan los siguientes resultados:

| Economia -Venta Semanal | Alta | Mediana | Baja | Total |
|-------------------------|------|---------|------|-------|
| PICO                    | 20   | 7       | 3    | 30    |
| ABAJO                   | 30   | 40      | 30   | 100   |
| SUBIENDO                | 20   | 8       | 2    | 30    |
| BAJANDO                 | 30   | 5       | 5    | 40    |
| TOTAL                   | 100  | 60      | 40   | 200   |

Realice la prueba de chi-cuadrado para validar la independencia del punto del ciclo de negocios en que se encuentre la economía.

# Ejercicio 2: prueba de chi cuadrada de homogeneidad

En un estudio a los habitos de television de los ninos, se selecciona una muestra aleatoria de 300 personas (100 ninos y 200 ninas). A cada infante se le pregunta que es mejor the lone ranger, sesame street o los simpson. Y se obtienen los siguientes resultados.

|       | Lone Ranger | Sesame Street | The Simpsons | Total |
|-------|-------------|---------------|--------------|-------|
| Boys  | 50          | 30            | 20           | 100   |
| Girls | 50          | 80            | 70           | 200   |
| Total | 100         | 110           | 90           | 300   |

Realice la prueba de chi-cuadrado para validar la homogeneidad. Las preferencias de los hombres difieren significativa de las ninas, usa 0.05 de nivel de significancia.

# Ejercicio 3: prueba de chi cuadrada de bondad ajuste

La compañia acme de juguetes imprime tarjetas de beisbol. La compañia dice que 30% de sus cartas son novatos, el 60% veteranos no estrellas y el otro 10% son veteranos estrellas.

Suponga que se toma un muestreo aleatorio de 100 cartas que tiene 50 novatos, 45 veteranos y 5 estrellas. Va acorde a lo que dice Acme?. Usa un nivel de significancia del 0.05

# Solucion de los Ejercicios


# Ejercicio 1: prueba de chi cuadrada de independencia

Para ver si la venta de chips de silicio son independientes del punto del ciclo de negocios en que se encuentre la economía del país se han recogido las ventas semanales de una empresa y datos acerca de la economía del país, y se reportan los siguientes resultados:

| Economia -Venta Semanal | Alta | Mediana | Baja | Total |
|-------------------------|------|---------|------|-------|
| PICO                    | 20   | 7       | 3    | 30    |
| ABAJO                   | 30   | 40      | 30   | 100   |
| SUBIENDO                | 20   | 8       | 2    | 30    |
| BAJANDO                 | 30   | 5       | 5    | 40    |
| TOTAL                   | 100  | 60      | 40   | 200   |

Realice la prueba de chi-cuadrado para validar la independencia del punto del ciclo de negocios en que se encuentre la economía.


### Solucion Nicolas

Entonces, para realizar la prueba de chi-cuadrada para validar la independencia de ambos eventos, establecemos que:

* Pa = Proporciones de ventas altas en economia pico 
* Pm = Proporcion de ventas medianas en economia pico 
* Pb = Proporcion de ventas bajas en economia pico 

Analogamente con los demas estados de la economia. Procedemos a establecer las hipotesis.

$$H_0: Pa = Pm = Pb$$ <center>(Las ventas son independentes del ciclo economico)
$$H_1: Pa , Pm , Pb$$  (Las ventas NO son independentes del ciclo economico)<center>

**Obtener Grados de Libertad**

GDL = (núm. Renglones – 1)(núm. Columnas - 1) = (4-1)(3-1) = 6

In [5]:
# definir los grados
GDL = 6

# Generar data frame con datos
datos = pd.DataFrame(index = ['PICO', 'ABAJO', 'SUBIENDO', 'BAJANDO', 'TOTAL ECONOMIA'],
                     columns = ['Alta', 'Mediana', 'Baja', 'Total'])
datos.iloc[0,:-1] = [20,7,3]
datos.iloc[1,:-1] = [30,40,30]
datos.iloc[2,:-1] = [20,8,2]
datos.iloc[3,:-1] = [30,5,5]
datos.loc['TOTAL ECONOMIA'] = datos.sum(axis = 0)
datos['Total'] = datos.sum(axis = 1)
datos

Alta Mediana Baja  Total
PICO             20       7    3   30.0
ABAJO            30      40   30  100.0
SUBIENDO         20       8    2   30.0
BAJANDO          30       5    5   40.0
TOTAL ECONOMIA  100      60   40  200.0

In [6]:
# encontrar la observacion esperada y la observada 
P = datos.iloc[:4,3]/datos.iloc[-1,-1]
f_o = datos.loc[datos.index!='TOTAL ECONOMIA',datos.columns!='Total'].values.flatten()
f_e = np.concatenate([datos.iloc[-1,:-1]*P[0],datos.iloc[-1,:-1]*P[1],datos.iloc[-1,:-1]*P[2],datos.iloc[-1,:-1]*P[3]])

f_e, f_o

(array([15.0, 9.0, 6.0, 50.0, 30.0, 20.0, 15.0, 9.0, 6.0, 20.0, 12.0, 8.0],
       dtype=object),
 array([20, 7, 3, 30, 40, 30, 20, 8, 2, 30, 5, 5], dtype=object))

In [7]:
# Metodo de chi cuadrado
x2 = st.chisquare(f_o,f_e,ddof=GDL)
print('Valor de chi cuadrado = ',list(x2)[0],',p-value de la prueba=',list(x2)[1])

Ji = st.chi2.ppf(q = 0.9,df=GDL)
print('Estadístico de Ji = ',Ji)

Valor de chi cuadrado =  34.59722222222222 ,p-value de la prueba= 1.810412864466741e-06
Estadístico de Ji =  10.644640675668422


### **Conclusion**

Se rechaza la hipotesis nula, es decir las ventas de chips de silicio **NO** son independientes del punto del ciclo de negocios en que se encuentre la economía del país. Se llego a esta conclusion debido a que el p_value de la prueba de chi-cuadrada fue de 1.810412864466741e-06, al ser muy pequeña siendo menor que un nivel de significancia de 0.1 o 0.05, se rechaza la hipotesis, ademas el valor del chi cuadrado fue de aproximadamente 34.59, superando por mucho el estadistico de chi de 10.64. 

### Solucion Doris

H0 = la venta de chips es INDEPENDIENTE al momento del ciclo económico donde se encuentra.

H alt = la venta de chips es DEPENDIENTE al momento del ciclo económico donde se encuentra.

In [3]:

data_C = pd.DataFrame(index=['En_Pico','En_PBajo_ABAJO','En_PBajo_SUBIENDO','En_PBajo_BAJANDO','total_e'],
                    columns=['Alta','Mediana','Baja','total'])
data_C.iloc[0,:-1] = [20,7,3]
data_C.iloc[1,:-1] = [30,40,30]
data_C.iloc[2,:-1] = [20,8,2]
data_C.iloc[3,:-1] = [30,5,5]

data_C['total'] = data_C.sum(axis=1)
data_C.loc['total_e',:] = data_C.sum(axis=0)
data_C

Alta Mediana Baja  total
En_Pico             20       7    3   30.0
En_PBajo_ABAJO      30      40   30  100.0
En_PBajo_SUBIENDO   20       8    2   30.0
En_PBajo_BAJANDO    30       5    5   40.0
total_e            100      60   40  200.0

In [5]:
P_C = data_C.iloc[:4,3]/data_C.iloc[-1,-1]

f_o_C = data_C.loc[data_C.index!='total_e',data_C.columns!='total'].values.flatten()


f_e_C = np.concatenate([data_C.iloc[-1,:-1]*P_C[0],data_C.iloc[-1,:-1]*P_C[1],data_C.iloc[-1,:-1]*P_C[2],data_C.iloc[-1,:-1]*P_C[3]])
f_e_C,f_o_C

(array([15.0, 9.0, 6.0, 50.0, 30.0, 20.0, 15.0, 9.0, 6.0, 20.0, 12.0, 8.0],
       dtype=object),
 array([20, 7, 3, 30, 40, 30, 20, 8, 2, 30, 5, 5], dtype=object))

In [6]:
#### Método chi cuadrado
F_obse_C = f_o_C
F_espe_C = f_e_C

x2_C = st.chisquare(F_obse_C,F_espe_C,ddof=6)
print('Valor de chi cuadrado = ',list(x2_C)[0],',p-value de la prueba=',list(x2_C)[1])

Ji_C = st.chi2.ppf(q = 0.9,df=6)
print('Estadístico de Ji = ',Ji_C)
# st.t.interval

Valor de chi cuadrado =  34.59722222222222 ,p-value de la prueba= 1.810412864466741e-06
Estadístico de Ji =  10.644640675668422


Rechazamos la H0 entonces aceptamos la Hipótesis alternativa de que la venta de chips es DEPENDIENTE al momento del ciclo económico donde se encuentra.

# Ejercicio 2: prueba de chi cuadrada de homogeneidad

En un estudio a los habitos de television de los ninos, se selecciona una muestra aleatoria de 300 personas (100 ninos y 200 ninas). A cada infante se le pregunta que es mejor the lone ranger, sesame street o los simpson. Y se obtienen los siguientes resultados.

|       | Lone Ranger | Sesame Street | The Simpsons | Total |
|-------|-------------|---------------|--------------|-------|
| Boys  | 50          | 30            | 20           | 100   |
| Girls | 50          | 80            | 70           | 200   |
| Total | 100         | 110           | 90           | 300   |

Realice la prueba de chi-cuadrado para validar la homogeneidad. Las preferencias de los hombres difieren significativa de las ninas, usa 0.05 de nivel de significancia.

### Solucion Nicolas

Las hipotesis que se definen son las siguientes:

*Hipotesis Nulas*  

H0: A los niños les gustan lone ranger =  A las niñas les gustan lone ranger  
H0: A los niños les gustan sesame street =  A las niñas les gustan lone ranger  
H0: A los niños les gustan sesame street =  A las niñas les gustan los simpsons

H1: No se cumple al menos una de las hipotesis nulas.

*Obtener grados de libertad*

GDL = (2-1)(3-1) = 2

In [8]:
# Definir grados de libertad
GDL2 = 2

# Crear data frame con datos
datos2 = pd.DataFrame(index = ['Boys', 'Girls', 'Total 1'],
                      columns = ['Lone Ranger', 'Sesame Street', 'The Simpsons', 'Total _'])
datos2.iloc[0,:-1] = (50,30,20)
datos2.iloc[1,:-1] = (50,80,70)
datos2['Total _'] = datos2.sum(axis = 1)
datos2.loc['Total 1'] = datos2.sum(axis = 0)
datos2


Lone Ranger Sesame Street The Simpsons  Total _
Boys             50            30           20    100.0
Girls            50            80           70    200.0
Total 1         100           110           90    300.0

In [9]:
# encontrar la observacion esperada y la observada 
P2 = datos2.iloc[:2,3]/datos2.iloc[-1,-1]
f_o2 = datos2.loc[datos2.index!='Total 1',datos2.columns!='Total _'].values.flatten()
f_e2 = np.concatenate([datos2.iloc[-1,:-1]*P2[0],datos2.iloc[-1,:-1]*P2[1]])

f_o2, f_e2

(array([50, 30, 20, 50, 80, 70], dtype=object),
 array([33.33333333333333, 36.666666666666664, 30.0, 66.66666666666666,
        73.33333333333333, 60.0], dtype=object))

In [10]:
# Metodo de chi cuadrada
x2 = st.chisquare(f_o2,f_e2,ddof=GDL2)
print('Valor de chi cuadrado = ',list(x2)[0],',p-value de la prueba=',list(x2)[1])

Ji = st.chi2.ppf(q = 0.95,df=GDL2)
print('Estadístico de Ji = ',Ji)

Valor de chi cuadrado =  19.31818181818182 ,p-value de la prueba= 0.00023495405224903583
Estadístico de Ji =  5.991464547107979


### **Conclusion**

Se rechaza la hipotesis nula, es decir las preferencias de los niños **NO** son homogeneas respecto a las de las niñas (son significativamente diferentes). Se llego a esta conclusion debido a que el p_value de la prueba de chi-cuadrada fue de 0.0023,y al ser muy pequeña con respecto al nivel de significancia pre-establecido de 0.05, se rechaza la hipotesis, ademas el valor del chi cuadrado fue de aproximadamente 19.3181, superando al estadistico de chi de 5.991464. 

### Solucion Doris

#### HOMOGENEIDAD
H0 = la proporción de niños que prefieren el Llanero Solitario es idéntica a la proporción de niñas. Del mismo modo, para los otros programas

H alternativa = almenos una de las poblaciones no cumple con la H0

Si p-value es menor que el nivel de significancia rechazamos H0

GDL = 2

In [7]:
data_H = pd.DataFrame(index=['chicos','chicas','total_N'],
                      columns=['lone_ranger','plaza_sesamo','los_simpson','total'])
data_H.iloc[0,:-1] = [50,30,20,]
data_H.iloc[1,:-1] = [50,80,70]
data_H['total'] = data_H.sum(axis=1)
data_H.loc['total_N',:] = data_H.sum(axis=0)
data_H

lone_ranger plaza_sesamo los_simpson  total
chicos           50           30          20  100.0
chicas           50           80          70  200.0
total_N         100          110          90  300.0

In [9]:
P_H = data_H.iloc[0,3]/data_H.iloc[-1,-1]
f_o_H = data_H.loc[data_H.index!='total_N',data_H.columns!='total'].values.flatten()
f_e_H = np.concatenate([data_H.iloc[-1,:-1]*P_H,data_H.iloc[-1,:-1]*(1-P_H)])

f_e_H,f_o_H

(array([33.33333333333333, 36.666666666666664, 30.0, 66.66666666666667,
        73.33333333333334, 60.00000000000001], dtype=object),
 array([50, 30, 20, 50, 80, 70], dtype=object))

In [10]:
#### Método chi cuadrado
F_obse_H = f_o_H
F_espe_H = f_e_H

x2_H = st.chisquare(F_obse_H,F_espe_H,ddof=2)
print('Valor de chi cuadrado = ',list(x2_H)[0],',p-value de la prueba=',list(x2_H)[1])

Ji_H = st.chi2.ppf(q = 0.95,df=2)
print('Estadístico de Ji = ',Ji_H)
# st.t.interval

Valor de chi cuadrado =  19.318181818181824 ,p-value de la prueba= 0.00023495405224903575
Estadístico de Ji =  5.991464547107979


RECHAZAMOS H0 que dice que la proporción de niños que prefieren el Llanero Solitario es idéntica a la proporción de niñas. Del mismo modo, para los otros programas

ENTONCES ACEPTAMOS LA ALTERNATIVA 


# Ejercicio 3: prueba de chi cuadrada de bondad ajuste

La compañia acme de juguetes imprime tarjetas de beisbol. La compañia dice que 30% de sus cartas son novatos, el 60% veteranos no estrellas y el otro 10% son veteranos estrellas.

Suponga que se toma un muestreo aleatorio de 100 cartas que tiene 50 novatos, 45 veteranos y 5 estrellas. Va acorde a lo que dice Acme?. Usa un nivel de significancia del 0.05

### Solucion Nicolas

*Definir Hipotesis*

H0 = La proporcion de como se distribuye cada carta es de 30% novatos, 60% veteranos, y 10% estrellas.

H1 = La proporcion se distribuye de forma diferente a lo que establece Acme

*Encontrar Grados de Libertad*

GDL = (k - 1); donde k es el numero de clasificaciones
GDL = 3 - 1 = 2


In [11]:
# Definir grados de libertad
GDL3 = 2

# Definir datos iniciales
n = 100 # tamano de la muestra
esp = np.array([0.30,0.60,0.10])
muestreo = np.array([50,45,5])

In [12]:
# encontrar las observaciones esperadas y las observadas
f_e3 = esp * n
f_o3 = muestreo

f_e3, f_o3

(array([30., 60., 10.]), array([50, 45,  5]))

In [13]:
# Metodo de chi cuadrada
x2 = st.chisquare(f_o3,f_e3,ddof=GDL3)
print('Valor de chi cuadrado = ',list(x2)[0],',p-value de la prueba=',list(x2)[1])

Ji = st.chi2.ppf(q = 0.95,df=GDL3)
print('Estadístico de Ji = ',Ji)

Valor de chi cuadrado =  19.583333333333336 ,p-value de la prueba= nan
Estadístico de Ji =  5.991464547107979


### **Conclusion**

Se rechaza la hipotesis nula, es decir las cartas de beisbol **NO** son se distribuyen de la forma que dice Acme Toy (de acuerdo al muestreo tomado). Se llego a esta conclusion debido a que el p_value de la prueba de chi-cuadrada fue de casi 0,y al ser muy pequeña con respecto al nivel de significancia pre-establecido de 0.05, se rechaza la hipotesis, ademas el valor del chi cuadrado fue de aproximadamente 19.58, superando al estadistico de chi de 5.991464. 

### Solucion Doris

#### BONDAD DE AJUSTE

H0 = La proporción de novatos, veteranos y Estrellas es del 30%, 60% y 10%, respectivamente.

H alternativa =Al menos una de las proporciones en la hipótesis nula es falsa.

Si p-value es menor que el nivel de significancia rechazamos H0


In [21]:
datos_B = pd.DataFrame(index = ['F_O','F_E'],
                      columns = ['Novatos', 'Veteranos', 'Estrellas', 'Total'])
datos_B.iloc[0,:-1] = (50,45,5)
datos_B.iloc[1,:-1] = (0.30,0.60,0.10)

datos_B['Total'] = datos_B.sum(axis = 1)

N = 100

datos_B

Novatos Veteranos Estrellas  Total
F_O      50        45         5  100.0
F_E     0.3       0.6       0.1    1.0

In [28]:
f_E_E=datos_B.iloc[1,:-1]*N
F_O = datos_B.iloc[0,:-1]

In [31]:
x2_B = st.chisquare(F_O,f_E_E,ddof=3)
print('Valor de chi cuadrado = ',list(x2_B)[0],',p-value de la prueba=',list(x2_B)[1])

Ji_B = st.chi2.ppf(q = 0.95,df=3)
print('Estadístico de Ji = ',Ji_B)

Valor de chi cuadrado =  19.583333333333336 ,p-value de la prueba= nan
Estadístico de Ji =  7.814727903251179


La Chi^2 es 0 lo que podría indicar una incorcondancia o bien que rechacemos nuestra H0.